js/plugins/googleai/src/index.ts

```diff
@@ -161,48 +176,159 @@ async function resolver(
   actionName: string,
   options?: PluginOptions
 ) {
-  // TODO: also support other actions like 'embedder'
   switch (actionType) {
     case 'model':
-      await resolveModel(ai, actionName, options);
+      resolveModel(ai, actionName, options);
+      break;
+    case 'embedder':
+      resolveEmbedder(ai, actionName, options);
       break;
     default:
     // no-op
   }
 }
 
-async function resolveModel(
+function resolveModel(ai: Genkit, actionName: string, options?: PluginOptions) {
+  const modelRef = gemini(actionName);
+  defineGoogleAIModel({
+    ai,
+    name: modelRef.name,
+    apiKey: options?.apiKey,
+    baseUrl: options?.baseUrl,
+    info: {
+      ...modelRef.info,
+      label: `Google AI - ${actionName}`,
+    },
+    debugTraces: options?.experimental_debugTraces,
+  });
+}
+
+function resolveEmbedder(
   ai: Genkit,
   actionName: string,
   options?: PluginOptions
 ) {
-  if (actionName.includes('gemini')) {
-    const modelRef = gemini(actionName);
-    defineGoogleAIModel({
-      ai,
-      name: modelRef.name,
-      apiKey: options?.apiKey,
-      baseUrl: options?.baseUrl,
-      info: {
-        ...modelRef.info,
-        label: `Google AI - ${actionName}`,
-      },
-      debugTraces: options?.experimental_debugTraces,
-    });
+  defineGoogleAIEmbedder(ai, `googleai/${actionName}`, {
```

Comment:
This if actionName is not already known, this defines an embedder with text only capability. I thought you mentioned that you wanted the generic model/embedder to "allow everything"? Shouldn't you modify the generic embedder definition as well?

Reply:
Done. made generic embedder multimodal.